### ISSUE TITLE
OneHotEncoder Python Extension is Broken

### ISSUE THREAD

**Issue by voganrc** [combust/mleap#667]
In scikit-learn's OneHotEncoder, unknown labels are represented as a row of all zeros:

https://github.com/scikit-learn/scikit-learn/blob/4755ae76d2df10bbf41bc93fb7083b0142ef1044/sklearn/preprocessing/_encoders.py#L220

Whereas in MLeap, unknown labels get their own column appended onto the one-hot encoded matrix:

https://github.com/combust/mleap/blob/3ee879fddb9d134c16ced719e2b0a3668f3071e2/mleap-core/src/main/scala/ml/combust/mleap/core/feature/OneHotEncoderModel.scala#L26

This means that in order to replicate the behavior of the scikit-learn OneHotEncoder, we have to drop this extra column. The way to do this in MLeap is to include `drop_last=True` in the bundle serialization.

But because the point of the OneHotEncoder Python extension is to allow users to choose the `drop_last` param:

https://github.com/combust/mleap/blob/d703c8dd5e7b93c7f6c8b45fa6c9aadb9c026a92/python/mleap/sklearn/extensions/data.py#L57

That class is useless. The serialized `drop_last` needs to always be `True`, because we always have to remove the invalid column from the MLeap implementation. User's don't actually have a choice whether they want to drop the last feature column.

**Comment by lucagiovagnoli** [comment]
Thanks for opening this issue. We can tackle this small sub-problem here rather than in #664 as that one is already huge

**Comment by ancasarb** [comment]
@voganrc @lucagiovagnoli Yes, the recent changes in OneHotEncoder/OneHotEncoderEstimator have been a bit of a nightmare in terms of maintaining backwards compatibility. Could you please take a look at https://github.com/combust/mleap/pull/677, which removes this and attempts to fix scikit-learn onehotencoder by serializing dropLast=true for parity? Thank you!
